elide-async/src/main/java/com/yahoo/elide/async/hooks/TableExportHook.java

```diff
@@ -15,29 +15,69 @@
 import com.yahoo.elide.core.security.ChangeSpec;
 import com.yahoo.elide.core.security.RequestScope;
 
+import org.apache.http.client.utils.URIBuilder;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URISyntaxException;
 import java.util.Optional;
 
 /**
  * LifeCycle Hook for execution of TableExpoer.
  */
+@Slf4j
 public class TableExportHook extends AsyncAPIHook<TableExport> {
+    private boolean baseURLPresent = true;
 
-    public TableExportHook (AsyncExecutorService asyncExecutorService, Integer maxAsyncAfterSeconds) {
+    public TableExportHook (AsyncExecutorService asyncExecutorService, Integer maxAsyncAfterSeconds,
+            boolean baseURLPresent) {
         super(asyncExecutorService, maxAsyncAfterSeconds);
+        this.baseURLPresent = baseURLPresent;
     }
 
     @Override
     public void execute(Operation operation, TransactionPhase phase, TableExport query, RequestScope requestScope,
             Optional<ChangeSpec> changes) {
         AsyncExecutorService service = getAsyncExecutorService();
-        TableExporter exporter = new TableExporter(service.getElide(), requestScope.getApiVersion(),
-                requestScope.getUser());
-        TableExportThread queryWorker = new TableExportThread(query, service.getResultStorageEngine(), exporter);
+
+        // TODO: Make it a Bean
+        TableExporter exporter = new TableExporter(service.getElide());
+
+        String downloadURL = retrieveDownloadURL(requestScope.getBaseUrlEndPoint());
+        TableExportThread queryWorker = new TableExportThread(query, exporter, downloadURL,
```

Comment:
The AsyncExecutorService should not have a function called isSkipCSVHeader.  That is specific configuration for TableExport.